cudax/include/cuda/experimental/__memory_resource/cuda_memory_pool.h

```diff
@@ -0,0 +1,265 @@
+  //! @param __device_id The device id of the device the stream pool is constructed on.
+  //! @param __pool_properties Optional, additional properties of the pool to be created
+  explicit cuda_memory_pool(const int __device_id, cuda_memory_pool_properties __properties = {})
+      : __pool_handle_(__create_cuda_mempool(__device_id, __properties))
+  {}
+
+  //! @brief Disables construction from a plain `cudaMemPool_t`. We want to ensure clean ownership semantics
+  cuda_memory_pool(::cudaMemPool_t) = delete;
+
+  cuda_memory_pool(cuda_memory_pool const&)            = delete;
+  cuda_memory_pool(cuda_memory_pool&&)                 = delete;
+  cuda_memory_pool& operator=(cuda_memory_pool const&) = delete;
+  cuda_memory_pool& operator=(cuda_memory_pool&&)      = delete;
+
+  //! @brief Destroys the \c cuda_memory_pool by releasing the internal ``cudaMemPool_t``.
+  ~cuda_memory_pool() noexcept
+  {
+    _CCCL_ASSERT_CUDA_API(::cudaMemPoolDestroy, "~cuda_memory_pool() failed to destroy pool", __pool_handle_);
+  }
+
+  //! @brief Equality comparison with another cuda_memory_pool
+  //! @return true if the stored ``cudaMemPool_t`` are equal
+  _CCCL_NODISCARD constexpr bool operator==(cuda_memory_pool const& __rhs) const noexcept
+  {
+    return __pool_handle_ == __rhs.__pool_handle_;
+  }
+
+#    if _CCCL_STD_VER <= 2017
+  //! @brief Inequality comparison with another cuda_memory_pool
+  //! @return true if the stored ``cudaMemPool_t`` are not equal
+  _CCCL_NODISCARD constexpr bool operator!=(cuda_memory_pool const& __rhs) const noexcept
+  {
+    return __pool_handle_ != __rhs.__pool_handle_;
+  }
+#    endif // _CCCL_STD_VER <= 2017
+
+  //! @brief Equality comparison with a cudaMemPool_t
+  //! @param __rhs A cudaMemPool_t
+  //! @return true if the stored ``cudaMemPool_t`` is equal to \p __rhs
+  _CCCL_NODISCARD_FRIEND constexpr bool operator==(cuda_memory_pool const& __lhs, ::cudaMemPool_t __rhs) noexcept
+  {
+    return __lhs.__pool_handle_ == __rhs;
+  }
+
+#    if _CCCL_STD_VER <= 2017
+  //! @copydoc cuda_memory_pool::operator==(cuda_memory_pool const&, ::cudaMemPool_t)
+  _CCCL_NODISCARD_FRIEND constexpr bool operator==(::cudaMemPool_t __lhs, cuda_memory_pool const& __rhs) noexcept
+  {
+    return __rhs.__pool_handle_ == __lhs;
+  }
+
+  //! @copydoc cuda_memory_pool::operator==(cuda_memory_pool const&, ::cudaMemPool_t)
+  _CCCL_NODISCARD_FRIEND constexpr bool operator!=(cuda_memory_pool const& __lhs, ::cudaMemPool_t __rhs) noexcept
+  {
+    return __lhs.__pool_handle_ != __rhs;
+  }
+
+  //! @copydoc cuda_memory_pool::operator==(cuda_memory_pool const&, ::cudaMemPool_t)
+  _CCCL_NODISCARD_FRIEND constexpr bool operator!=(::cudaMemPool_t __lhs, cuda_memory_pool const& __rhs) noexcept
+  {
+    return __rhs.__pool_handle_ != __lhs;
+  }
+#    endif // _CCCL_STD_VER <= 2017
+
+  //! @brief Returns the underlying handle to the CUDA memory pool
+  _CCCL_NODISCARD constexpr cudaMemPool_t pool_handle() const noexcept
```

Comment:
The standard library generally calls this `native_handle`, do we maybe want to follow that convention? The event APi already uses that terminology, so we also have a precedent in cudax.

Reply:
Renamed